gnome-base/gnome: bump meta ebuilds to 47.3-ish

EvaSDK opened the pull request:
Gnome releng did some changes post 47.2 release and stopped producing easy to crawl lists of packages. Nevertheless, did to libgit2-glib dependency issues that stopped portage from resolving properly world updates (probably for fractal + gitg + git-absorb), I had to find a way to reduce the amount of packages to update that could cause trouble. I wanted to update gnome but found out the meta ebuilds hadn't been updated in a while. Here are bumps of the meta ebuilds with versions as close as possible to what Gnome published for 47.3. Since 47.4 is due in less that 2 weeks, I figured it would be good enough for now.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

EvaSDK committed:
gnome-base/gnome-core-libs: add 47.3

Signed-off-by: Gilles Dartiguelongue <gilles.dartiguelongue@esiee.org>

EvaSDK committed:
gnome-base/gnome-core-apps: add 47.3

Signed-off-by: Gilles Dartiguelongue <gilles.dartiguelongue@esiee.org>

EvaSDK committed:
gnome-base/gnome-extra-apps: add 47.3

Signed-off-by: Gilles Dartiguelongue <gilles.dartiguelongue@esiee.org>

EvaSDK committed:
gnome-base/gnome: add 47.3

ppc64 keyword dropped as it is not present in gnome-shell-47. Could not
find a ticket to link to.

Signed-off-by: Gilles Dartiguelongue <gilles.dartiguelongue@esiee.org>

cfuga commented on gnome-base/gnome-extra-apps/gnome-extra-apps-47.3.ebuild:
```diff
@@ -0,0 +1,73 @@
+	>=gnome-base/dconf-editor-45.0.1
+	>=mail-client/evolution-3.54.3
+	>=gnome-extra/gnome-tweaks-46.1
+	>=gnome-extra/nautilus-sendto-3.8.6
```
Please, drop ```>=gnome-extra/nautilus-sendto-3.8.6``` dependency: ```gnome-base/nautilus``` doesn't require it since version 43.beta.

https://gitlab.gnome.org/GNOME/nautilus/-/issues/928
https://gitlab.gnome.org/GNOME/nautilus/-/issues/2126

Also, the project is archived upstream.

EvaSDK replied:
Done.

pacho2 commented:
Hi! Thanks for bringing this up. 

I would also consider adding gnome-clocks to gnome-extra-apps (https://bugs.gentoo.org/656946).

Do you have any idea about where we can retrieve the updated GNOME apps lists from upstream? 
For "core" apps and libs I found the following upstream lists:
https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/master/elements/core/meta-gnome-core-developer-tools.bst
https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/master/elements/core/meta-gnome-core-mobile.bst
https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/master/elements/core/meta-gnome-core-os-services.bst
https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/master/elements/core/meta-gnome-core-shell.bst

But I have no idea about the equivalent of "extra" meta.

cfuga commented:
> Hi! Thanks for bringing this up.
> 
> I would also consider adding gnome-clocks to gnome-extra-apps (https://bugs.gentoo.org/656946).
> 
> Do you have any idea about where we can retrieve the updated GNOME apps lists from upstream? For "core" apps and libs I found the following upstream lists: https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/master/elements/core/meta-gnome-core-developer-tools.bst https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/master/elements/core/meta-gnome-core-mobile.bst https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/master/elements/core/meta-gnome-core-os-services.bst https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/master/elements/core/meta-gnome-core-shell.bst
> 
> But I have no idea about the equivalent of "extra" meta.

I figured out the required information.

Until version 3.26, the GNOME releng team kept a list with core libs/apps separated from extra apps. 

https://gitlab.gnome.org/GNOME/releng/-/blob/eb6197439d4942c63b43c4f9d39ecf5603a6af20/tools/versions

Gentoo has two metapackages, based on that list: ```gnome-base/gnome-core-apps```, containing the ```core``` labeled apps, and ```gnome-base/gnome-extra-apps```, managing the ```apps``` labeled apps.

But the ```apps``` label was removed since GNOME 3.28.

https://gitlab.gnome.org/GNOME/releng/-/commit/f0da599c70cd65d5bec0f747af78a022297ed887
https://gitlab.gnome.org/GNOME/releng/-/blob/33afb4214a29adb27811aafda732f92fda4f7ba3/tools/versions-stable

Currently, the metapackage ```gnome-base/gnome-extra-apps``` is a Gentoo's historical/inherited list of apps that doesn't relate to any upstream info.

e-luks commented:
Looking at https://apps.gnome.org and https://gitlab.gnome.org/Teams/Circle/-/blob/main/data/apps.json?ref_type=heads and  https://gitlab.gnome.org/GNOME/gnome-app-list/-/tree/main/data?ref_type=heads, the newest entry is the Keypunch app. Note while  Keypunch is not directly present in gnome-build-meta, gnome-app-list is referenced inside gnome-build-meta in gnome-build-meta/elements/core/gnome-software.bst and gnome-build-meta/elements/core/gnome-software.bst

e-luks commented on gnome-base/gnome-core-apps/gnome-core-apps-47.3.ebuild:
```diff
@@ -0,0 +1,54 @@
+	>=media-video/totem-43.1
+	|| (
+		>=x11-terms/gnome-terminal-3.54.3
+		>=gui-apps/gnome-console-47.1
```
I'd really like to include >=x11-terms/ptyxis-47.6 in here, which AFAIK is the default on Ubuntu 25.04 and Fedora 41.

zyxhere replied:
Ptyxis is not a core GNOME app (yet?) so please don't add it here

zyxhere replied:
Both console and terminal are core apps (not intentional but whats done is done)

e-luks replied:
As far as i can see only console is in core apps, terminal is not (https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/master/elements/core/meta-gnome-core-apps.bst).

EvaSDK replied:
I think the discussion here is mostly resolved. Whether gnome-console should be first in the list over gnome-terminal is a maintainer decision. I tested it shortly and it was behaving fine but I personally switched to ghostty so I don't mind either way.

zyxhere replied:
`gnome-console` is the modern GTK4 + libadwaita so fits nicely with other GNOME apps :)

e-luks commented:
When talking just about gnome core or circle apps, the list can be fetched as json from the api: https://apps.gnome.org/_api/apps.json, each entry has gnome: "core" or gnome: "circle"

e-luks commented:
I made an obviously imperfect package comparison between the current ebuilds and the bst files with removed package categories and suffixes.

The point that i'm trying to make here is that i am worried about medium to long term maintainability, i think we should stay closer to upstream wording and meta package categories

git diff --no-index <(cat gentoo-core-apps-47.3.ebuild) <(cat meta-gnome-core-shell.bst) 
diff --git a/dev/fd/63 b/dev/fd/62
``` diff
--- a/dev/fd/63
+++ b/dev/fd/62
@@ -1,21 +1,25 @@
 adwaita-icon-theme
-eog
-evince
-evolution-data-server
-gcr
-gedit
+gdm
 glib-networking
+gnome-backgrounds
 gnome-bluetooth
 gnome-color-manager
-gnome-console
-gnome-contacts
 gnome-control-center
+gnome-desktop
+gnome-initial-setup
 gnome-keyring
+gnome-menus
+gnome-remote-desktop
 gnome-session
 gnome-settings-daemon
-gnome-terminal
-gnome-text-editor
+gnome-shell
+gnome-tour
 gnome-user-docs
-nautilus
-totem
-yelp
+gnome-user-share
+gsettings-desktop-schemas
+gvfs-daemon`
+mutter`
+orca
+rygel
+sushi
+tecla
```

git diff --no-index <(cat gnome-extra-apps-47.3.ebuild) <(cat meta-gnome-core-apps.bst)
``` diff
 baobab
-cheese
-dconf-editor
+decibels
 epiphany
-evolution
-file-roller
-five-or-more
-four-in-a-row
+evince
 gnome-calculator
 gnome-calendar
 gnome-characters
-gnome-chess
+gnome-clocks
+gnome-connections
+gnome-console
+gnome-contacts
 gnome-disk-utility
 gnome-font-viewer
-gnome-klotski
-gnome-mahjongg
-gnome-mines
+gnome-logs
+gnome-maps
 gnome-music
-gnome-nibbles
-gnome-photos
-gnome-robots
-gnome-sudoku
+gnome-software
 gnome-system-monitor
-gnome-taquin
-gnome-tetravex
-gnome-tweaks
-gnome-user-share
+gnome-text-editor
 gnome-weather
-hitori
-iagno
-lightsoff
-nautilus-sendto
-quadrapassel
-seahorse
```

robszy commented:
When I can expect gnome 47 to be in portage ?

lukilou commented:
Yep, assuming we have 48 release... if some help with 47 release needed I can give some help.

zyxhere commented:
Sent with Proton Mail secure email.

On Saturday, 22 March 2025 at 3:38 PM, Luca Peeters ***@***.***> wrote:

> lukilou left a comment (gentoo/gentoo#40658)
> 
> Yep, assuming we have 48 release... if some help with 47 release needed I can give some help.
> 
GNOME 47 is already in ::gentoo just needs to be stablized or you can just use ~amd64 in `make.conf`

zyxhere commented:
@EvaSDK any interest in also bumping `gnome-base/gnome-light` too?

robszy commented:
> Sent with Proton Mail secure email.
> On Saturday, 22 March 2025 at 3:38 PM, Luca Peeters ***@***.***> wrote: lukilou left a comment ([gentoo/gentoo#40658](https://github.com/gentoo/gentoo/pull/40658)) Yep, assuming we have 48 release... if some help with 47 release needed I can give some help.
> GNOME 47 is already in ::gentoo just needs to be stablized or you can just use ~amd64 in `make.conf`

Sorry, to be precise when it will stabilize ?

zyxhere commented:
On Saturday, 22 March 2025 at 5:25 PM, robert ***@***.***> wrote:

> robszy left a comment (gentoo/gentoo#40658)
> 
> > Sent with Proton Mail secure email.
> > On Saturday, 22 March 2025 at 3:38 PM, Luca Peeters @.***> wrote: lukilou left a comment (gentoo/gentoo#40658) Yep, assuming we have 48 release... if some help with 47 release needed I can give some help.
> > GNOME 47 is already in ::gentoo just needs to be stablized or you can just use ~amd64 in `make.conf`
> 
> Sorry, to be precise when it will stabilize ?

You can open an issue about it on bugzilla :)

robszy commented:
It is stabilizing from 2025-01-27 19:49:12 UTC so it should be good now :)

EvaSDK committed:
gnome-base/gnome-extra-apps: drop gnome-extra/nautilus-sendto

Feature is now provided by xdg-desktop-portal. gnome-shell depends on
sys-apps/xdg-desktop-portal-gnome for this. Also the project is
archived.

Refs: https://github.com/gentoo/gentoo/pull/40658#discussion_r1963943123
Signed-off-by: Gilles Dartiguelongue <gilles.dartiguelongue@esiee.org>

EvaSDK commented on gnome-base/gnome-extra-apps/gnome-extra-apps-47.3.ebuild:
```diff
@@ -0,0 +1,73 @@
+	tracker? (
+		>=app-misc/tracker-3.6.0
+		>=app-misc/tracker-miners-3.6.2
+		>=media-gfx/gnome-photos-44.0
```
gnome-photos being part of core or not does not really matter in gnome-extras-apps. It would in gnome-core-apps. I will let gnome maintainers decide whether to retain it or not considering it's release cycle is probably not inline with Gnome anymore but there are no replacements.

EvaSDK committed:
gnome-base/gnome-light: add 47.3

Dropped ppc64 due to missing keywords in dependencies.

Signed-off-by: Gilles Dartiguelongue <gilles.dartiguelongue@esiee.org>

zyxhere commented:
> It is stabilizing from 2025-01-27 19:49:12 UTC so it should be good now :)

Its really not so you should open one about it

robszy commented:
> > It is stabilizing from 2025-01-27 19:49:12 UTC so it should be good now :)
> 
> Its really not so you should open one about it

So what are the issues it it is really not stabilized ?

zyxhere commented:
On Saturday, 3 May 2025 at 2:14 PM, robert ***@***.***> wrote:

> robszy left a comment (gentoo/gentoo#40658)
> 
> > > It is stabilizing from 2025-01-27 19:49:12 UTC so it should be good now :)
> > 
> > Its really not so you should open one about it
> 
> So what are the issues it it is really not stabilized ?
> 
I really don't know probably because no one has opened a stabilisation bug about it

EvaSDK committed:
gnome-base/gnome-light: cosmetic alignment with gnome-base/gnome ebuild

Signed-off-by: Gilles Dartiguelongue <gilles.dartiguelongue@esiee.org>

robszy commented:
I submitted bug https://bugs.gentoo.org/955830 but i don't think it will help :(

zyxhere commented:
On Sunday, 11 May 2025 at 2:29 PM, robert ***@***.***> wrote:

> robszy left a comment (gentoo/gentoo#40658)
> 
> I submitted bug https://bugs.gentoo.org/955830 but i don't think it will help :(
> 

What you can do to help is take a look at https://bugs.gentoo.org/946484 and add the necessary packages to package list for stabilisation of gnome-base/gnome
Also change the summary to gnome-base/gnome: stablereq

robszy commented:
Isnt this PR actually also for that problem eg to match list of packages with 47.3 version of gnome ?

EvaSDK commented:
I'm not sure why the bot didn't assign the PR properly. In any case, I got tired of waiting and I moved on to 47.6 in my overlay.

pacho2 commented:
Sorry for the delay, Gilles.

Can you please update the PR to the 47.6 version from your overlay? I will try to look into it soon.

By the way, maybe you could also submit a PR for your loupe ebuild (I saw that you already packaged it in your overlay and adapted the metas for it as an alternative to eog)

Thanks a lot
